utils/utils.py

```diff
@@ -129,7 +131,57 @@ def format_date(
 		if date.year != base.year:
 			date_format += " %Y"
 	# include the time if it is not an all day event and the time is different from the base
-	if not all_day and date.strftime("%d%b%I:%M:%S%p") != base.strftime("%d%b%I:%M:%S%p"):
+	if not all_day and date.strftime("%d%b%I%M%S%p") != base.strftime("%d%b%I%M%S%p"):
```

Comment:
Does this have fixed number of digits? Make sure that things like 12/1 and 1/21 don't both get mapped to 121 and collide. It's probably not the best idea to compare dates by stringifying them. Date objects should be == comparable